src/Spice86.Core/Emulator/OperatingSystem/IDosPathResolver.cs

```diff
@@ -0,0 +1,88 @@
+namespace Spice86.Core.Emulator.OperatingSystem;
+
+using Spice86.Core.Emulator.OperatingSystem.Structures;
+
+using System.Collections.Generic;
+
+/// <summary>
+/// Translates DOS filepaths to host file paths, and vice-versa.
+/// </summary>
+public interface IDosPathResolver {
+    /// <summary>
+    /// The map between host paths and the root of DOS drives.
+    /// </summary>
+    IDictionary<char, MountedFolder> DriveMap { get; }
+
+    /// <summary>
+    /// The current DOS drive in use.
+    /// </summary>
+    char CurrentDrive { get; }
+
+    /// <summary>
+    /// The full host path to the folder used by DOS as the current folder.
+    /// </summary>
+    string CurrentHostDirectory { get; }
+
+    /// <summary>
+    /// Create a relative path from the current host directory to another. Paths will be resolved before calculating the difference.
+    /// Default path comparison for the active platform will be used (OrdinalIgnoreCase for Windows or Mac, Ordinal for Unix).
+    /// </summary>
+    /// <param name="hostPath">The destination path.</param>
+    /// <returns>The relative path or <paramref name="hostPath"/> if the paths don't share the same root.</returns>
+    string GetHostRelativePathToCurrentDirectory(string hostPath);
+
+    /// <summary>
+    /// Converts the DOS path to a full host path.<br/>
+    /// </summary>
+    /// <param name="dosPath">The file name to convert.</param>
+    /// <param name="convertParentOnly">if true, it will try to find the case sensitive match for only the parent of the path</param>
+    /// <returns>A string containing the full file path in the host file system, or <c>null</c> if nothing was found.</returns>
+    string? ToHostCaseSensitiveFullName(string dosPath, bool convertParentOnly);
+
+    /// <summary>
+    /// Returns the full host file path, including casing.
+    /// </summary>
+    /// <param name="dosFilePath">The DOS file path.</param>
+    /// <returns>A string containing the host file path, or <c>null</c> if not found.</returns>
+    string? TryGetFullHostFileName(string dosFilePath);
```

Comment:
What is the difference between these two methods?
Or, how would a user of this interface know which one to use?

Reply:
This is still the most confusing part remaining.

Reply:
Fixed.